TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -914,7 +914,7 @@ public virtual void EndSentence(SentenceEnd end, bool bCheck)
         /// </summary>
         public virtual void CheckStartSentenceLastStatement()
         {
-            if (LastEnteredNode != null)
+            if (LastEnteredNode is Statement)
```

Comment:
ok with this change.
Can you also add a check in `StartParagraph(ParagraphHeader header)`:
```C#
       public static bool NeedDump = true;
       public virtual void StartParagraph(ParagraphHeader header)
        {
           //pseudo code
            if (CurrentNode is Sentence)
            { 
                  Exit(CurrentNode);
                  //Dump only once to avoid too much logs in our system
                  if (NeedDump)
                  {
                      Dump all sources with _codeElementsLines;
                      NeedDump = false;
                  }
            }
        }
```

Reply:
Done